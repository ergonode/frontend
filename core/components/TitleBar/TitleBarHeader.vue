/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <div class="title-bar-header">
        <slot name="prepend" />
        <h1
            class="title-bar-header__title"
            v-text="title" />
        <slot name="append" />
    </div>
</template>

<script>

export default {
    name: 'TitleBarHeader',
    props: {
        title: {
            type: String,
            required: true,
        },
    },
};
</script>

<style lang="scss" scoped>
    .title-bar-header {
        display: grid;
        grid-auto-flow: column;
        align-items: center;
        grid-column-gap: 16px;

        &__title {
            color: $GRAPHITE_DARK;
            font: $FONT_MEDIUM_24_32;
        }
    }
</style>
