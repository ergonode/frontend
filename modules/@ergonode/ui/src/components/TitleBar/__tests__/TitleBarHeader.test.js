/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
import TitleBarHeader from '@UI/components/TitleBar/TitleBarHeader';
import {
    mount,
} from '@vue/test-utils';

describe('TitleBar/TitleBarHeader', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = mount(TitleBarHeader, {
            propsData: {
                title: 'test',
                isBreadcrumb: true,
            },
        });
    });

    it('Component is named well', () => {
        expect(typeof TitleBarHeader.name).toBe('string');
        expect(TitleBarHeader.name).toEqual('TitleBarHeader');
    });

    it('Header has breadcrumb', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
