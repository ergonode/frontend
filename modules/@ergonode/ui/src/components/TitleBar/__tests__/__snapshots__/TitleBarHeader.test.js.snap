// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBar/TitleBarHeader Header has breadcrumb 1`] = `
<div data-cy="title-bar-header" class="title-bar-header" isbreadcrumb="true">
  <h1 class="title-bar-header__title">test</h1>
</div>
`;
